src/chunk_tuple_routing.c

```diff
@@ -137,6 +141,10 @@ ts_chunk_tuple_routing_find_chunk(ChunkTupleRouting *ctr, Point *point)
 				Chunk *compressed_chunk =
 					ts_cm_functions->compression_chunk_create(compressed_ht, chunk);
 				ts_chunk_set_compressed_chunk(chunk, compressed_chunk->fd.id);
+
+				/* mark chunk as partial unless completely new chunk */
+				if (!new_chunk)
+					ts_chunk_set_partial(chunk);
```

Comment:
It's confusing that it's now marked as partial from two different functions under different conditions. This function is returning a ChunkInsertState, can it set the required flags there instead?

Reply:
Not sure this results in more clarity. This is an existing chunk that gets columnstore created because of direct compress so it needs to be marked as partial

Reply:
I changed it to be in one place